File: 1/lua/p2.lua
local depths = {}
local count = 0

for line in io.lines() do
  table.insert(depths, tonumber(line))
end

for i = 1, #depths - 3 do
  local a = depths[i] + depths[i + 1] + depths[i + 2]
  local b = depths[i + 1] + depths[i + 2] + depths[i + 3]

  if b - a > 0 then
    count = count + 1
  end
end

print('Number of increases', count)

File: 1/js/p2.js
const readline = require('readline');

const depths = [];
let count = 0;

const rl = readline.createInterface({
  input: process.stdin
});

rl.on('line', (l) => {
  depths.push(parseInt(l));
});

rl.on('close', () => {
  for (let i = 0; i < depths.length - 3; i++) {
    const a = depths[i] + depths[i + 1] + depths[i + 2];
    const b = depths[i + 1] + depths[i + 2] + depths[i + 3];

    if (b - a > 0) {
      count++;
    }
  }

  console.log(`Number of increases: ${count}`);
});
